crates/core/src/client/helpers.rs

```diff
@@ -232,17 +232,25 @@ pub fn decode_eth_send_transaction_return(call_result: &[FieldElement]) -> Resul
     Ok(segmented_result)
 }
 
-pub fn decode_signature_from_tx_calldata(calldata: &[FieldElement]) -> Result<Signature, DataDecodingError> {
+pub fn decode_signature_and_to_address_from_tx_calldata(
```

Comment:
I fully agree with those statements.
How would `StarknetTransactionDecoder` work?

Agree with both doing it in `StarknetTransactionDecoder` (rename it to KakarotTransactionDecoder probably since here we decode kkt calldata format), 

and also could do it in StarknetTransaction.

Should this be in a subsequent PR, to not burden danilo's current flow

Reply:
Yes no worries let's make an issue for it